stratified-split/script.whizzml

```diff
@@ -0,0 +1,82 @@
+;; Transposes the matrix: [[a b] [c d]] to [[a c] [b d]]
+
+(define (_transpose m)
+  (apply map list m))
+
+;; Creates the each dataset by sampling and using out-of-bag
+
+(define (sample-dataset ds-id rate oob)
+  (create-and-wait-dataset {"sample_rate" rate
+                            "origin_dataset"  ds-id
+                            "out_of_bag" oob
+                            "seed" "bigml"}))
+
+;; Splits the original dataset in two complementary datasets using
+;; a given sample rate
+
+(define (split-dataset ds-id rate)
+  (list (sample-dataset ds-id rate false)
+        (sample-dataset ds-id rate true)))
+
+;; Deletes resources ignoring errors
+(define (safe-delete id)
+  (try (delete id)
+       (catch e
+         (log-info (str "Error deleting resource " id " ignored"))
+         false)))
+
+;; Check that the objective field exists and is a categorical field
+
+(define (check-cat fields objective-field)
+  (let (field (find-field fields objective-field))
+    (when (not (find-field fields objective-field))
+      (raise {"message" (str "The objective field cannot be found in this"
+                             " dataset")
```

Comment:
i would perhaps print also `objective-field` in the error message, just for quick reference when seeing the error...